code/segmentedtickerplant/stplog.q

```diff
@@ -0,0 +1,98 @@
+// Utilites for periodic tp logging in stp process
+
+.proc.loadf[raze .proc.params`stpconfig]
+
+\d .stplg
+
+d:.z.d
+t:tables`.
+
+// Create stp log directory
+// Log structure `:stplogs/date/tabname_time
+createdld:{[name;date]
+  .os.md dldir::hsym`$getenv[`TORQHOME],(,/)"/stplogs/",string name,date
+ };
+
+// Default stp mode is tabperiod
+// TP log rolled periodically (default 1 hr), 1 log per table
+logname.tabperiod:{[dir;tab;logfreq;dailyadj]
+  ` sv(hsym dir;`$string[tab],ssr[;;""]/[-13_string logfreq xbar .z.p+dailyadj;":.D"])
+ };
+
+// Standard TP mode - to be added
+logname.none:{[dir;tab;logfreq;dailyadj]}
+
+// Custom mode, configured by user - to be added
+logname.custom:{[dir;tab;logfreq;dailyadj]}
+
+// Live logs and handles to logs for each table
+currlog:([tbl:`symbol$()]logname:`symbol$();handle:`int$())
+
+openlog:{[multilog;dir;tab;logfreq;dailyadj]
+  lname:logname[multilog][dir;tab;logfreq;dailyadj];
+  if[not type key lname;.[lname;();:;()]];
+  h:hopen lname;
+  `.stplg.currlog upsert (tab;lname;h);
+  if[addmeta;.stpm.updmeta[`open;lname;tab]];
+ };
+
+// Error log for failed updates in error mode
+openlogerr:{[dir]
+  lname:` sv(hsym dir;`$"errdatabase",string .z.d);
+  if[not type key lname;.[lname;();:;()]];
```

Comment:
you probably want to error trap all these disk i/o ops of setting / opening the log file. Never know what can happen

Reply:
Still need error traps around these